diag_manager/fms_diag_object.F90

```diff
@@ -872,4 +880,27 @@ subroutine dump_diag_obj( filename )
   call mpp_error( FATAL, "You can not use the modern diag manager without compiling with -Duse_yaml")
 #endif
 end subroutine
+
+!> @brief Copies input data to output data with proper type if the input data is present
+!! else sets the output data to a given value val if it is present.
+!! If the value val and the input data are not present, the output data is untouched.
+subroutine real_copy_set(out_data, in_data, val)
```

Comment:
This routine is more generic and should be in diag_utils I think.  Routines in this module should be type bound procedures.

Reply:
I wanted to replace lines https://github.com/NOAA-GFDL/FMS/blob/bb25a07716dfcf3d097a5c88c31ac0d03a9904e0/diag_manager/diag_manager.F90#L1872-L1884 with this routine.

Reply:
I can move the routine to diag_utils_mod.